noir-projects/aztec-nr/aztec/src/event/event_interface.nr

```diff
@@ -14,50 +14,136 @@ use dep::protocol_types::{
-    // TODO(#14565): This variant requires for tagging to also be constrained, as it is otherwise useless.
-    pub CONSTRAINED_ENCRYPTION: u8,
+    /// Useful when we trust the sender or the sender is incentivized to deliver the message. See combination 2
+    /// above for details.
+    pub UNCONSTRAINED_OFFCHAIN: u8,
+
+    /// Unconstrained encryption + unconstrained tagging + private log stream
+    ///
+    /// As mentioned above, this combination is not meaningful but we keep it here as it's currently used in the
+    /// codebase.
+    /// TODO: Consider dropping this.
+    pub UNCONSTRAINED_ONCHAIN: u8,
 }
 
-pub global PrivateLogContent: PrivateLogContentConstraintsEnum = PrivateLogContentConstraintsEnum {
-    NO_CONSTRAINTS: 1,
-    CONSTRAINED_ENCRYPTION: 2,
-    // TODO: add constrained tagging and constrained handshaking
+pub global MessageDelivery: MessageDeliveryEnum = MessageDeliveryEnum {
+    CONSTRAINED_ONCHAIN: 1,
+    UNCONSTRAINED_OFFCHAIN: 2,
+    UNCONSTRAINED_ONCHAIN: 3,
 };
 
-/// Emits an event in a private log, encrypting it such that only `recipient` will learn of its contents. The log will
-/// be tagged using a shared secret between `sender` and `recipient`, so that `recipient` can efficiently find the log.
+/// Emits an event that can be delivered either via private logs or offchain messages, with configurable encryption and
+/// tagging constraints.
 ///
-/// The `constraints` value determines what parts of this computation will be constrained. See the documentation for
-/// each value in `PrivateLogContentConstraintsEnum` to learn more about the different variants.
-pub fn emit_event_in_private_log<Event>(
+/// # Arguments
+/// * `event` - The event to emit
+/// * `context` - The private context to emit the event in
+/// * `recipient` - The address that should receive this event
+/// * `delivery_mode` - Controls encryption, tagging, and delivery constraints. Must be a compile-time constant.
+///   See `MessageDeliveryEnum` for details on the available modes.
+pub fn emit_event_in_private<Event>(
     event: Event,
     context: &mut PrivateContext,
     recipient: AztecAddress,
-    constraints: u8,
+    delivery_mode: u8,
 )
 where
     Event: EventInterface + Serialize,
 {
-    // This function relies on `constraints` being a constant in order to reduce circuit constraints when unconstrained
-    // usage is requested. If `constraints` were a runtime value then performance would suffer.
-    assert_constant(constraints);
+    // This function relies on `delivery_mode` being a constant in order to reduce circuit constraints when unconstrained
+    // usage is requested. If `delivery_mode` were a runtime value then performance would suffer.
+    assert_constant(delivery_mode);
+
+    // The following maps out the 3 dimensions across which we configure message delivery.
+    let unconstrained_encryption = delivery_mode != MessageDelivery.CONSTRAINED_ONCHAIN;
```

Comment:
I think this would read better with equality and ORs, as you'd see more clearly what each variant does. E.g.

```noir
let unconstrained_encryption = delivery_mode == MessageDelivery.UNCONSTRAINED_OFFCHAIN | UNCONSTRAINED_ONCHAIN;
```

it also avoid the double negation in the `!=` and `unconstrained_enc`. It'd likely be even better if we had 


```noir
let constrained_encryption = delivery_mode == MessageDelivery.CONSTRAINED_ONCHAIN;
```

Reply:
Other than that these booleans are nice.

Reply:
Changing that to `constrained_encryption` indeed [looks nicer](https://github.com/AztecProtocol/aztec-packages/pull/16126/commits/6b9306c163928c730d6cb879f639be0c10c6146c). Thanks